src/main/java/core/basesyntax/service/operations/BalanceOperation.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        storageFruits.put(transaction.getFruit(), transaction.getQuantity());
```

Comment:
It seems like 'storageFruits' is a direct reference to a storage collection. This violates the encapsulation principle. Consider accessing the storage through a service or a DAO (Data Access Object) that abstracts the underlying storage operations.